### ISSUE TITLE
Use vault with TLS enabled

### ISSUE THREAD

**Issue by PurpleBenny** [rajanadar/VaultSharp#34]
Hi rajanadar,
I have vault on remote server and vault TLS is enable. So when I want communicate with vault i need apply certificate. However in vault sharp i cannot find tool how i say it 'use this certificate'. I'm beginner with vault and with SSL comunication so it's possible that I forgot something.

Thank you

**Comment by rajanadar** [comment]
hi @PurpleBenny 

Thanks for using the library. For TLS (assuming auth backend, and not secret backend) this is how you access Vault using an Auth Backend token.

https://github.com/rajanadar/VaultSharp/blob/master/README.md#certificate-tls-authentication-backend

`
var clientCertificate = new X509Certificate2(certificatePath, certificatePassword, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.PersistKeySet);

IAuthenticationInfo certificateAuthenticationInfo = new CertificateAuthenticationInfo(mountPoint, clientCertificate);
IVaultClient vaultClient = VaultClientFactory.CreateVaultClient(vaultUriWithPort, certificateAuthenticationInfo);

// any operations done using the vaultClient will use the vault token/policies mapped to the client certificate.
`

Basically we pre-configure Vault with information about what client certificate maps to what Vault Policies. So when we use the Cert Auth and provide a client certificate, Vault can map it to the appropriate policies and give you a vault token with those privileges.

and if you are looking for Cert Secret Backend info (not Auth backend), here is the link for it:

https://github.com/rajanadar/VaultSharp/blob/master/README.md#pki-certificates-secret-backend

jfyi. VaultSharp 0.9.x is in rapid development, and expected middle of Jan 2018.

- Cheers!

**Comment by PurpleBenny** [comment]
Thank you @rajanadar for very quick response. In the end there was a problem in another issue, which does not concern with vaultsharp absolutely.

Thanks for good job on this great library

**Comment by rajanadar** [comment]
np. thanks @PurpleBenny
